codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -282,6 +285,16 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                 return;
             }
 
+            boolean isInformational = false;
+            if ((!connection.isServer() &&
+                    (isInformational = HttpStatusClass.valueOf(headers.status()) == INFORMATIONAL)
+                    && stream.isInformationalHeadersReceived()) ||
```

Comment:
Yes you are right ... I wasn't aware that infinite could be received/sent. Thanks for pointing this out.

Are you aware of any practical use cases for more than 1 info response?